op-batcher/batcher/espresso.go

```diff
@@ -221,28 +220,33 @@ func (l *BatchSubmitter) espressoBatchLoadingLoop(ctx context.Context, wg *sync.
 }
 
 type BlockLoader struct {
-	prevSyncStatus  *eth.SyncStatus
-	lastQueuedBlock *eth.L2BlockRef
-	batcher         *BatchSubmitter
+	queuedBlocks   []eth.L2BlockRef
+	prevSyncStatus *eth.SyncStatus
+	batcher        *BatchSubmitter
 }
 
-func (l *BlockLoader) Reset(ctx context.Context) {
+func (l *BlockLoader) reset(ctx context.Context) {
 	l.prevSyncStatus = nil
-	l.lastQueuedBlock = nil
+	l.queuedBlocks = nil
 	l.batcher.clearState(ctx)
+	l.batcher.streamer.Reset()
 }
 
 func (l *BlockLoader) EnqueueBlocks(ctx context.Context, blocksToQueue inclusiveBlockRange) {
+	l.batcher.Log.Info("Loading and queueing blocks", "range", blocksToQueue)
 	for i := blocksToQueue.start; i <= blocksToQueue.end; i++ {
 		block, err := l.batcher.fetchBlock(ctx, i)
-		if errors.Is(err, ErrReorg) {
-			l.batcher.Log.Warn("Found L2 reorg", "block_number", i)
-			l.Reset(ctx)
-			break
-		} else if err != nil {
+		if err != nil {
 			l.batcher.Log.Warn("Failed to fetch block", "err", err)
 			break
 		}
+
+		if len(l.queuedBlocks) > 0 && block.ParentHash() != l.queuedBlocks[len(l.queuedBlocks)-1].Hash {
+			l.batcher.Log.Warn("Found L2 reorg", "block_number", i)
+			l.reset(ctx)
+			break
+		}
```

Comment:
Is this for debugging purposes? I thought reorgs would be handled in `espressoBatchLoadingLoop`, not here.

Reply:
We don't have this particular check there. This function is the first place we actually get the block and so can check whether it correctly extends the parent.